### PR TITLE
ci: Add github codespaces devcontainer

### PR DESCRIPTION
**Pull request by daroga0002**
# PR o'clock

## Description

This PR is adding devcontainer definition. This will allow to develop code using GitHub codespaces. 

It is installing in codespaces:
- terraform
- tflint
- terraform-docs
- pre-commit

### Checklist

- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by daroga0002** [comment]
@bryantbiggs  for me this seems be beneficial for simpler contribution using codespaces, but what are your thoughts?

**Comment by bryantbiggs** [comment]
I'll be honest - I've never used devcontainers/codespaces so I can't really comment 😅 - I don't know if anyone else is that could maybe chime in. on thought though, if you do want to use them we could add the `.devcontainer` path to `.gitignore` so that you have it locally but its not committed. otherwise I will leave it up to others that might know more about this

**Comment by antonbabenko** [comment]
I have been using various gitpod/codespaces and less popular ones (like Theia) but I have not yet figured out how to bring those configs as part of the repositories.

Let's merge this PR merged but keep in mind that if there will be issues reported by the users we will have a limited experience to support them because it is out of scope for the module repository.

PS: Please add ["betajob.modulestf"](https://marketplace.visualstudio.com/items?itemName=betajob.modulestf) to the list of pre-installed VS Code Extensions. You will hear more about it in 2022. :)

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"settings": {
+		"terraform.languageServer": {
+			"enabled": true,
+			"args": []
+		}
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"hashicorp.terraform",
```
Comment (antonbabenko) [inline .devcontainer/devcontainer.json:22]: Please add [betajob.modulestf](https://marketplace.visualstudio.com/items?itemName=betajob.modulestf) to this list.

**Pull request closed without merging**